migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,210 @@
+		newConfig := create_v2_report_config(reportConfigProto)
+
+		// create notifier config for report_configuration_notifier table
+		notifierConfig := create_notifier(reportConfigProto)
+		newConfig.Notifiers = append(newConfig.Notifiers, notifierConfig)
+
+		// add notifier to notifier_configurations_notifiers
+		reportNotifierGorm, err := updatedSchema.ConvertNotifierConfigurationFromProto(notifierConfig, 0, newConfig.GetId())
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", notifierConfig)
+		}
+		reportNotifiers = append(reportNotifiers, reportNotifierGorm)
+
+		// assign version to 2 to new copy and version to 1 in original so that they are not re-created during migration
+		newConfig.Version = 2
+		reportConfigProto.Version = 1
+		// convert report config proto back to gorm model
+		convertedGormNewConfig, err := updatedSchema.ConvertReportConfigurationFromProto(newConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", newConfig)
+		}
+		convertedGormReportConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", reportConfigProto)
+		}
+		convertedReportConfigs = append(convertedReportConfigs, convertedGormNewConfig, convertedGormReportConfig)
+
+		// create report snapshot for last run report job
+		reportSnapshot := create_report_snapshot(reportConfigProto, newConfig)
+		if reportSnapshot != nil {
+			// convert report snapshot to GORM
+			reportSnapshotGORM, err := updatedSchema.ConvertReportSnapshotFromProto(reportSnapshot)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert from proto %+v", reportSnapshot)
+			}
+			reportSnapshots = append(reportSnapshots, reportSnapshotGORM)
+		}
+		if len(convertedReportConfigs) == batchSize {
+			if err = db.Clauses(clause.OnConflict{UpdateAll: true}).Model(updatedSchema.CreateTableReportConfigurationsStmt.GormModel).Create(&convertedReportConfigs).Error; err != nil {
+				return errors.Wrap(err, "failed to upsert converted report configs")
+			}
+			if err = dbSnapshot.Clauses(clause.OnConflict{UpdateAll: true}).Model(updatedSchema.CreateTableReportSnapshotsStmt.GormModel).Create(&reportSnapshots).Error; err != nil {
+				return errors.Wrap(err, "failed to upsert converted report snapshots")
+			}
+
+			if err = dbNotifier.Clauses(clause.OnConflict{UpdateAll: true}).Model(updatedSchema.CreateTableNotifiersStmt.GormModel).Create(&reportNotifiers).Error; err != nil {
+				return errors.Wrap(err, "failed to upsert converted report notifier configurations")
+			}
+			convertedReportConfigs = convertedReportConfigs[:0]
+			reportSnapshots = reportSnapshots[:0]
+			reportNotifiers = reportNotifiers[:0]
+		}
+	}
+	if err = db.Clauses(clause.OnConflict{UpdateAll: true}).Model(updatedSchema.CreateTableReportConfigurationsStmt.GormModel).Create(&convertedReportConfigs).Error; err != nil {
```

Comment:
Need to check the length of each array to make sure there are records to persist.  If by chance overall records / batchSize == 0, blindly passing an empty array will fail.